deploy/common/k8sbased.sh

```diff
@@ -727,7 +727,7 @@ function launch_sensor {
         helm_args+=(--set "allowNonstandardNamespace=true")
       fi
 
-      if [[ "$SENSOR_HELM_MANAGED" == "true" ]]; then
+      if [[ "$SENSOR_HELM_DEPLOY" == "true" ]]; then
```

Comment:
We already check the value of this variable on line 671 so this check would be useless.
I'd consult @SimonBaeumer who has explained the idea behind `helmManaged` to me multiple times but I still fail to grasp it 😅

Reply:
For some reason, I thought that when the `SENSOR_HELM_MANAGED` was introduced [here](https://github.com/stackrox/rox/pull/7778), `SENSOR_HELM_DEPLOY` did not exist yet, but on a second look it actually was.

Even if the separate `SENSOR_HELM_MANAGED` is actually needed here, something is wrong because removing it didn't make any test fail.

Reply:
The naming is confusing. `helmManaged=true`  means to upgrade Sensor a user must use `helm`.
If `helmManaged=false` it means that Central tracks the Sensor as `manual` installation method. This was done to enable auto-upgrades for users who used Helm to install Sensor, but don't want to upgrade Sensor via Helm.

Reply:
I am now confused Simon. We have different means to control the behavior of the upgrader and the only source of truth is the `managedBy` field. If there is another field doing the same, then it should be removed.

Reply:
@vikin91 If `helmManaged` is disabled, `managedBy` will be set to `MANAGER_TYPE_MANUAL`:
https://github.com/stackrox/stackrox/blob/master/image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl#L271

And at this point removing `helmManaged` would be breaking for customers who use it. We could at least document it.

@SimonBaeumer but does it make sense that the deploy script creates by default a Helm installed secured cluster, that is configured as not Helm managed? I found this quite confusing.

Reply:
@vikin91 ^what Vlad said

@vladbologa Good Q, I think we did it because of backward compatibility to CI runs. A lot of the code was introduced in the deploy scripts when we introduced the Operator and Helm (both shortly after each other). There was not time to clean-up or rethink our decisions, after the operator was finished we started ACSCS.
Imho you can be free in how to work with the deploy scripts and which goals you want to achieve with it.

Reply:
> If helmManaged is disabled, managedBy will be set to MANAGER_TYPE_MANUAL:
[master/image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl#L271](https://github.com/stackrox/stackrox/blob/master/image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl?rgh-link-date=2024-12-04T11%3A43%3A24Z#L271)

@vladbologa Does that work correctly for the operator? We don't have here a bool (helm or manual) but an enum (helm, operator, manual).

Reply:
@vikin91 `helmManaged` defaults to true, unless overridden. Since it's not exposed in the operator CSV, it's not possible to change its value when using the operator, so `managedBy` should always be `MANAGER_TYPE_KUBERNETES_OPERATOR` when using it.

Reply:
Unless you manually set `helmManaged` to `false` (which is not possible with the operator), the values of `helmManaged` and `managedBy` are determined by this:

https://github.com/stackrox/stackrox/blob/830cd40b682bd1c3f92ca5bd4a12e3455c6a8e77/image/templates/helm/stackrox-secured-cluster/internal/defaults/30-base-config.yaml.htpl#L12-L20

Reply:
Okay, I will not get too deep to the other values of the Helm Chart and trust you with the right selection. I only want to stress that the secured cluster upgrader relies only on the value of `managedBy` and it is read from the config map after everything is installed. All changes leading to this value being correct are fine for me.